src/Components/Web.JS/src/Platform/Circuits/CircuitStreamingInterop.ts

```diff
@@ -11,10 +12,46 @@ export function sendJSDataStream(connection: HubConnection, data: ArrayBufferVie
         try {
             let position = 0;
             let chunkId = 0;
+            let tmpData = new Uint8Array();
+            let reader: ReadableStreamDefaultReader | undefined = data instanceof DotNet.StreamWithLength ? data.stream.getReader() : undefined;
+
+            async function getNextChunk(): Promise<Uint8Array> {
+                if (reader === undefined) {
+                    throw new Error('Failed to get stream reader.');
+                }
+
+                let nextChunk: Uint8Array;
+
+                if (tmpData.length < chunkSize) {
+                    const readResult: ReadableStreamDefaultReadResult<any> = await reader.read();
```

Comment:
We can't specify size of chunk we'd like to read so we need to do a bit of legwork to get the chunk sizes to align with the rest of the streaming interop implementation (mainly abiding by SignalR message size constraints).

Reply:
Could we not operate directly over the file? It provides a `slice` method you could use?

Reply:
I'd considered `slice`, but opted to go the `blob.stream` route as `ReadableStream` may be something that we'd be able to use in other areas as well, whereas `blob.slice` would be File I/O specific.

Reply:
Fair enough

Reply:
Here's the `Blob.slice` based implementation: https://github.com/dotnet/aspnetcore/pull/33900/commits/fc45a1ddbf0593738f3c0c7838bcb002ce3401fd

Advantages of this approach is it's much simpler (no byte array manipulation to get around a bunch of edge conditions on piping between the file stream and the JS<>.NET interop stream. It also should [work fully in Safari](https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice#browser_compatibility) so there won't be any need for back compat and so on.